edb/pgsql/metaschema.py

```diff
@@ -5049,169 +5046,98 @@ def _generate_sql_information_schema() -> List[dbops.View]:
                 GROUP BY link.id, link.cardinality
                 HAVING link.cardinality = 'Many' OR COUNT(*) > 2
             )
-            SELECT link.id, obj_ty.schema_name,
+            SELECT link.id, obj_ty.schema_name, obj_ty.module_name,
                 CONCAT(obj_ty.table_name, '.', link.name) AS table_name
             FROM edgedb."_SchemaLink" link
             JOIN obj_ty ON obj_ty.id = link.source
             WHERE link.id IN (SELECT * FROM qualified_links)
         ) UNION ALL (
             -- multi properties
-            SELECT prop.id, obj_ty.schema_name,
+            SELECT prop.id, obj_ty.schema_name, obj_ty.module_name,
                 CONCAT(obj_ty.table_name, '.', prop.name) AS table_name
             FROM edgedb."_SchemaProperty" AS prop
             JOIN obj_ty ON obj_ty.id = prop.source
             WHERE prop.computable IS NOT TRUE
-              AND prop.internal IS NOT TRUE
-              AND prop.cardinality = 'Many'
+            AND prop.internal IS NOT TRUE
+            AND prop.cardinality = 'Many'
         ))
+        SELECT at.id, schema_name, table_name, sm.id as module_id, module_name
+        FROM all_tables at
+        JOIN edgedb."_SchemaModule" sm ON sm.name = module_name
+        WHERE schema_name not in ('cfg', 'sys', 'schema', 'std')
+        '''
+    )
+    # A few tables in here were causing problems, so let's hide them as an
+    # implementation detail.
+    # To be more specific:
+    # - following tables were missing from information_schema:
+    #   Link.properties, ObjectType.links, ObjectType.properties
+    # - even though introspection worked, I wasn't able to select from some
+    #   tables in cfg and sys
+
+    # For making up oids of schemas that represent modules
+    uuid_to_oid = dbops.Function(
```

Comment:
Hm, what do we think the odds of collision are? How big is oid?

Reply:
32-bit signed integers, so technically this can overflow.